core/src/test/java/com/onelogin/saml2/test/authn/AuthnResponseTest.java

```diff
@@ -1096,90 +1030,86 @@ public void testIsInValidSessionIndex() throws Exception {
 		assertEquals("The attributes have expired, based on the SessionNotOnOrAfter of the AttributeStatement of this Response", samlResponse.getError());
 	}
 
-	/**
-	 * Tests the isValid method of SamlResponse
-	 * Case: invalid SubjectConfirmation
-	 *
-	 * @throws Exception
-	 *
-	 * @see com.onelogin.saml2.authn.SamlResponse#isValid
-	 */
 	@Test
-	public void testIsInValidSubjectConfirmation() throws Exception {
```

Comment:
Split this one into separate tests for more clarity, and refactored the shared parts.